sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdContextRequestEncoder.java

```diff
@@ -26,7 +26,7 @@
      */
     @Override
     public boolean acceptOutboundMessage(final Object message) {
-        return message.getClass() == RntbdContextRequest.class;
+        return message instanceof RntbdContextRequest;
```

Comment:
Is it future proofing or a bug? 
Quick check doesn't show any sub-classes.

Reply:
Reverted in next iteration